## Objective

This project is part of the UMI normalization described in the README of [UmiNormalize](https://github.com/nirmalya-broad/UMINormalize). The basic algorithm of UMI normalization assumes that the reads from bacteria generated using [scDual-seq](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-017-1340-x) protocol represents underlying structure of bacterial transcripts. Since a bacterial transcript can span multiple genes, reads generated from that transcript can span multiple genes. Inference of the boundary of the original transcript is not straightforward and according to our knowledge, there is no available method to solve this problem. The first version of the UMI normalization collapsed those reads by making a relatively simple assumption:

<i>Reads from two different transcripts with the same attached UMI are separated by at least a predefined number of bases.</i>

In the [first version of UMI](https://www.nature.com/articles/s41598-019-55633-6), we used a fixed value of 500 for this gap between reads to infer the transcripts, which we estimated using basic histogram analysis of the read distribution. It provided reasonably accurate UMI normalization.

In this project, we attempt to model the gaps between the reads using a statistical approach. 

## Problem formulation

The library preparation of [scDual-seq](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-017-1340-x) uses hexamer priming in two different stages followed by PCR amplification. After careful inspection of the library protocol, we model the gap between consecutive reads under a single UMI as a mixture of three distributions. 

Let's denote the n<sup>th</sup> gap by x<sub>n</sub>. We can write x<sub>n</sub> as

prob(x<sub>n</sub>) = &pi;<sub>0</sub> &delta;<sub>0</sub>(x<sub>n</sub>) + &pi;<sub>1</sub> NB(x<sub>n</sub> | r<sub>1</sub>, p<sub>1</sub>) + &pi;<sub>2</sub> NB(x<sub>n</sub> | r<sub>2</sub>, p<sub>2</sub>)

The first density is a [Dirac Measure](https://en.wikipedia.org/wiki/Dirac_measure) which represents PCR based amplification since reads generated by PCR from a single transcript are separated by zero bases. The other two densities represent (a) gaps between two consecutive reads from a single transcript under a single UMI (b) gaps between two consecutive reads from two consecutive transcripts under a single UMI. For example, let there be two consecutive transcripts T<sub>1</sub> and T<sub>2</sub> labeled with the same UMI. Let R<sub>T<sub>1</sub></sub> denote the last read on the 3<sup>&prime;</sup> side of T<sub>1</sub> and R<sub>T<sub>2</sub></sub> denote the first read on the 5<sup>&prime;</sup> end of T<sub>2</sub>. Then this density model the gap between R<sub>T<sub>1</sub></sub> and R<sub>T<sub>2</sub></sub>. &pi;<sub>0</sub>, &pi;<sub>1</sub> and &pi;<sub>2</sub> denote the corresponding prior probabilities respectively.

Once, we estimate the parameters of the distribution (which include &pi;<sub>0</sub>, &pi;<sub>1</sub>, &pi;<sub>2</sub>,  r<sub>1</sub>, p<sub>1</sub>,  r<sub>2</sub> and p<sub>2</sub>), we can cluster the reads in that UMI.

Note that, (1) We initially attempted to use geometric distributions instead of negative binomial. However, geometric distribution was not able to model the over dispersion of the data according to the q-q plot. Instead we use negative binomial distribution which provides much better results (2)  Parameters estimated from the model depend on the distribution of reads. Hence, parameters obtained from different segment of the genome are different. For example, for a sample without ribosomal RNA depletion, the parameters might be different from the ones obtained from sparsely populated regions. (3) We assume that on a specific region on the genome, the reads are distributed identically and independently with respect to UMIs. Hence, we combine reads from all the UMIs while estimating the parameters.

## Method


